osc-server/src/main/java/org/osc/core/broker/util/network/NetworkSettingsApi.java

```diff
@@ -75,77 +45,87 @@ public NetworkSettingsDto getNetworkSettings() {
-            networkSettingsDto.setHostSubnetMask("");
-            networkSettingsDto.setHostDefaultGateway("");
-            networkSettingsDto.setHostDnsServer1("");
-            networkSettingsDto.setHostDnsServer2("");
+            networkSettingsDto.setHostSubnetMask(getIPv4LocalNetMask());
+            networkSettingsDto.setHostDefaultGateway(getDefaultGateway());
+            List<String> dns = getDNSSettings();
+            networkSettingsDto.setHostDnsServer1(dns.get(0));
+            networkSettingsDto.setHostDnsServer2(dns.get(1));
+
+        } catch (SocketException | UnknownHostException e) {
+            log.error("Failed to get host and/or ip address", e);
+        } catch (IOException e) {
+            log.error("Failed to get DNS Server(s)", e);
         }
 
-        return networkSettingsDto;
+    return networkSettingsDto;
     }
 
-    private String[] getDNSSettings() throws IOException {
-        String[] dns = { "", "" };
+    private List<String> getDNSSettings() throws IOException {
+
+        List<String> dns = new ArrayList<>();
+        int index = 0;
 
-        int i = 0;
         try (FileReader fileReader = new FileReader(this.NETWORK_RESOLV);
-             Scanner in = new Scanner(fileReader)) {
+            Scanner in = new Scanner(fileReader)) {
             while (in.hasNextLine()) {
                 String[] tokens = in.nextLine().split(" ");
                 String s = tokens[0];
                 if (s.startsWith("name") && tokens.length > 1) {
-                    dns[i++] = tokens[1];
-                    if (i > 2) {
+                    dns.add(tokens[1]);
+                    index++;
+                    if (index > 1) {
                         break;
                     }
                 }
             }
         }
 
-        return dns;
+    return dns;
     }
 
+    String getIPv4LocalNetMask() {
+
+        String netMask = "";
+        String[] cmd = { "/bin/sh", "-c", "ip addr show eth0 |grep inet|tr -s ' ' |cut -d' ' -f3" };
+        List<String> outlines = new ArrayList<>();
+
+        int exitCode = ServerUtil.execWithLog(cmd, outlines);
+        if (exitCode != 0) {
+            log.error("Encountered error during: {} execution", cmd);
+            return null;
+        }
+
+        for(String line:outlines) {
+            netMask = line;
+            break;
+        }
+
+    return netMask;
+    }
+
+    String getDefaultGateway() {
+
+        String defaultGateway = "";
+        String[] cmd = { "/bin/sh", "-c", "ip route|grep default|tr -s ' ' |cut -d' ' -f3" };
+        List<String> outlines = new ArrayList<>();
+
+        int exitCode = ServerUtil.execWithLog(cmd, outlines);
+        if (exitCode != 0) {
+            log.error("Encountered error during: {} execution", cmd);
+            return null;
+        }
+
+        for (String line:outlines) {
+            defaultGateway = line;
+            break;
+        }
+
+    return defaultGateway;
```

Comment:
@sudhirappaji @bmmukund this is still poorly indented. line 129 should align with ln 127. Please fix here an other places before merge.